src/employee/employee.service.ts

```diff
@@ -0,0 +1,51 @@
+import { IEmployeeRepository } from "./employee.repository";
+import { EmployeeType, EmployeeCreateType } from "./interfaces";
+import { IRepositories } from "../repositories";
+
+export interface IEmployeeService {
+  createEmployee: (employee: EmployeeCreateType) => Promise<EmployeeType>;
+  getAllEmployees: () => Promise<EmployeeType[]>;
+  getEmployeeById: (id: number) => Promise<EmployeeType | undefined>;
+}
+
+class EmployeeService implements IEmployeeService {
+  private readonly employeeRepository: IEmployeeRepository;
+  constructor({ employeeRepository }: IRepositories) {
+    this.employeeRepository = employeeRepository;
+  }
+
+  async createEmployee({
+    name,
+    surname,
+    githubAccount,
+    employmentType,
+  }: EmployeeCreateType) {
+    const user = await this.employeeRepository.findUnique(
+      "githubAccount",
+      githubAccount
+    );
+
+    if (user) {
+      throw new Error(
+        `User with githubAccount: ${githubAccount} already exist`
+      ); // TODO change to exceptions latter
+    }
+
+    return await this.employeeRepository.createEmployee({
+      name,
+      surname,
+      githubAccount,
+      employmentType,
+    });
+  }
+
+  async getAllEmployees() {
+    return await this.employeeRepository.getAllEmployees();
```

Comment:
Jeśli zwracasz promise i nie czekasz na jej wynik w catch, nie musisz korzystać ze słów async/await

Reply:
w tym wypadku ma kompletną rację, choć jest przypadek, gdzie może mieć to znaczenie https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise